src/bandersnatch/tests/test_mirror.py

```diff
@@ -1166,9 +1167,9 @@ async def test_cleanup_non_pep_503_paths(mirror: BandersnatchMirror) -> None:
     touch_files([mirror.webdir / "simple" / raw_package_name / "index.html"])
 
     mirror.cleanup = True
-    with mock.patch("bandersnatch.mirror.rmtree") as mocked_rmtree:
-        await mirror.cleanup_non_pep_503_paths(package)
-        assert mocked_rmtree.call_count == 1
+    await mirror.cleanup_non_pep_503_paths(package)
+    # TODO: add a mock here.
+    # we need a mock to mock delete behavior, but currently I cannot implement it.
```

Comment:
I feel here we just have to mock the two calls ...
```suggestion
     with mock.patch("bandersnatch.mirror.Path.unlink") as mocked_unlink, mock.patch("bandersnatch.mirror.Path.rmdir" as mocked_rmdir):
         await mirror.cleanup_non_pep_503_paths(package)
         assert mocked_unlink == X  # number you expect
         assert mocked_rmdir == 1  # Or number you expect here
```